packages/crml-cennzx-spot/src/CennzxSpot.ts

```diff
@@ -18,229 +18,202 @@ import {AssetId} from '@cennznet/types';
+   * @param assetSold The asset to sell
+   * @param assetBuy The asset to buy
+   * @param amountSell amount of trade asset 1 to sell
+   * @param minSale Min trade asset 2 to receive from sale (output)
+   */
+  assetSwapInput(
+    assetSold: AnyAssetId,
+    assetBought: AnyAssetId,
+    amountSell: AnyNumber,
+    minReceive: AnyNumber
+  ): SubmittableExtrinsic<'promise'> {
+    return this.api.tx.cennzxSpot.assetSwapInput(null, assetSold, assetBought, amountSell, minReceive);
+  }
+
+  /**
+   * Sell AssetSold, gain AssetBought as payback then transfer to recipient
+   * @param recipient - The address that receives the output asset
+   * @param assetSold The asset to sell
+   * @param assetBuy The asset to buy
+   * @param amountSell amount of trade asset to sell
+   * @param minSale Min core asset to receive from sale (output)
+   */
+  assetTransferInput(
+    recipient: AnyAddress,
+    assetSold: AnyAssetId,
+    assetBought: AnyAssetId,
+    amountSell: AnyNumber,
+    minReceive: AnyNumber
+  ): SubmittableExtrinsic<'promise'> {
+    return this.api.tx.cennzxSpot.assetSwapInput(recipient, assetSold, assetBought, amountSell, minReceive);
+  }
+
+  /**
+   * Query the total liquidity of an exchange pool
+   * @param assetId
+   */
+  get getTotalLiquidity(): QueryableTotalLiquidityBalance {
+    const _fn = this.api.derive.cennzxSpot.totalLiquidity as QueryableTotalLiquidityBalance;
+    _fn.at = this.api.derive.cennzxSpot.totalLiquidityAt;
+
+    return _fn;
+  }
+
+  /**
+   * Query the core asset id
+   */
+  get getCoreAssetId(): QueryableStorageEntry<'promise', AssetId> {
+    return this.api.query.cennzxSpot.coreAssetId as any;
+  }
+
+  /**
+   * Query liquidity balance for an account
+   * @param assetId The id of the asset
+   * @param address The address of the account
+   */
+  get getLiquidityBalance(): QueryableGetLiquidityBalance {
+    const _fn = this.api.derive.cennzxSpot.liquidityBalance as QueryableGetLiquidityBalance;
+    _fn.at = this.api.derive.cennzxSpot.liquidityBalanceAt;
+
+    return _fn;
+  }
+
+  /**
+   * Query balance for an exchange pool
+   * @param assetId The id of the asset
+   */
+  get getPoolAssetBalance(): QueryableGetPoolBalance {
+    const _fn = this.api.derive.cennzxSpot.poolAssetBalance as QueryableGetPoolBalance;
+    _fn.at = this.api.derive.cennzxSpot.poolAssetBalanceAt;
+
+    return _fn;
+  }
+
+  /**
+   * Query balance for an exchange pool
+   * @param assetId The id of the asset
+   */
+  get getPoolCoreAssetBalance(): QueryableGetPoolBalance {
+    const _fn = this.api.derive.cennzxSpot.poolCoreAssetBalance as QueryableGetPoolBalance;
+    _fn.at = this.api.derive.cennzxSpot.poolCoreAssetBalanceAt;
+
+    return _fn;
+  }
+
+  /**
+   * Query liquidity price for a core asset amount
+   * @param assetId The id of the asset
+   * @param coreAmount - the amount of core asset
+   */
+  get liquidityPrice(): QueryableGetLiquidityBalancePrice {
```

Comment:
Random note: It's weird that some methods are prefixed **get**`TheThingToQuery` while some are just `thingToQuery`
We could improve this